src/HotChocolate/Core/src/Types/Types/Descriptors/DirectiveArgumentDescriptor.cs

```diff
@@ -22,17 +23,23 @@ public class DirectiveArgumentDescriptor
             PropertyInfo property)
             : base(context)
         {
-            Definition.Name = context.Naming.GetMemberName(
-                property, MemberKind.DirectiveArgument);
-            Definition.Description = context.Naming.GetMemberDescription(
-                property, MemberKind.DirectiveArgument);
+            Definition.Name =
+                context.Naming.GetMemberName(property, DirectiveArgument);
+            Definition.Description =
+                context.Naming.GetMemberDescription(property, DirectiveArgument);
+
             Definition.Type = context.TypeInspector.GetInputReturnTypeRef(property);
             Definition.Property = property;
 
             if (context.TypeInspector.TryGetDefaultValue(property, out object defaultValue))
             {
                 Definition.RuntimeDefaultValue = defaultValue;
             }
+
+            if (context.Naming.IsDeprecated(property, out var reason))
```

Comment:
The spec sys:

```
deprecated fields on a type, arguments on a field, input fields on an input
 type, or values of an enum type.
 ```
 
 Directives cannot be used on directives this directives arguments cannot be deprecated.

Reply:
For reference:
spec/Section 3 -- Type System.md